### PR TITLE
feat(@vtmn/vue): add `VtmnModal` component

### PR DESCRIPTION
**Pull request by GaspardMathon**
## Changes description

- add `VtmnModal` component

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

### CODE REVIEW

File: packages/showcases/vue/stories/components/overlays/VtmnModal/VtmnModal.stories.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        in quia? Lorem ipsum dolor, sit amet consectetur adipisicing elit.
+      </div>
+    </template>
+    
+    <template v-slot:actions>
+      <div class="vtmn-modal_content_actions">
+        <VtmnButton variant="secondary">Action 1</VtmnButton>
+        <VtmnButton variant="primary">Action 2</VtmnButton>
+      </div>
+    </template>
```
Comment (GaspardMathon) [inline packages/showcases/vue/stories/components/overlays/VtmnModal/VtmnModal.stories.js:57]: @ThomasLaigneauDecathlon Is ti ok here to pass the parent div of the button actions ? Or is there another way to implement the slots ?

**Review by lauthieb — CHANGES_REQUESTED** [review]

**Review by ThomasLaigneauDecathlon — COMMENTED** [review]
Can we talk about the global implementation of this component ? Feel free to check this file and we can discuss about it ASAP.

https://github.com/dktunited/finance-front-check-and-go/blob/develop/src/components/base/BaseModal.vue

**Comment by lauthieb** [comment]
@ThomasLaigneauDecathlon yes for sure!
We tried to align a maximum with existing components in React & Svelte for your information.

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    props = reactive(props);
+
+    const handleClose = (event: Event) => {
+      return emit('close', (event.target as HTMLInputElement).value);
+    };
+
+    return {
+      classes: computed(() => ({
+        'vtmn-modal': true,
+        show: !props.animationDisabled,
```
Comment (ThomasLaigneauDecathlon) [inline packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:31]: I don't understand what you are trying to do with the show: !props.animationDisabled.

If it is a class you should add quotes around it. Maybe you want to create an other computed ref ?

**Review by ThomasLaigneauDecathlon — CHANGES_REQUESTED** [review]

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue (hunk 2 above)
Comment (lauthieb) [reply packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:31]: But @GaspardMathon I do agree with @ThomasLaigneauDecathlon for the fact that's it's weird to link "show" with the prop "animationDisabled", it's not the same purpose.

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks @GaspardMathon, just one change :)

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      type: Boolean as PropType<Boolean>,
+      default: false,
+    },
+  },
+  emits: ['close'],
+  setup(props, { emit }) {
+    props = reactive(props);
+
+    const handleClose = (event: Event) => {
+      return emit('close', (event.target as HTMLInputElement).value);
```
[the remark this answers is not in the document]
Comment (GaspardMathon) [reply packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:25]: ok thanks :)

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      classes: computed(() => ({
+        'vtmn-modal': true,
+      })),
+      handleClose,
+    };
+  },
+});
+</script>
+
+<template v-if="show">
```
Comment (ThomasLaigneauDecathlon) [inline packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:34]: Warning here, you can't use v-if or anything on the <template> opening tag of a component !
The v-if="show" should be use on the div juste after this line.

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  <div :class="classes" role="dialog" aria-modal="true" v-bind="$attrs">
+    <div
+      id="vtmn-modal-background"
```
Comment (ThomasLaigneauDecathlon) [inline packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:37]: Why adding this id ?

**Review by ThomasLaigneauDecathlon — CHANGES_REQUESTED** [review]

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue (hunk 4 above)
Comment (GaspardMathon) [reply packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:34]: ok thanks :)

File: packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import { computed, defineComponent, PropType, reactive } from 'vue';
+import { VtmnButton } from '../../index';
+
+export default /*#__PURE__*/ defineComponent({
+  name: 'VtmnModal',
+  components: { VtmnButton },
+  inheritAttrs: false,
+  props: {
+    open: {
+      type: Boolean as PropType<Boolean>,
```
Comment (lauthieb) [inline packages/sources/vue/src/components/overlays/VtmnModal/VtmnModal.vue:12]:
```suggestion
      type: Boolean as PropType<boolean>,
```

File: packages/showcases/vue/stories/components/overlays/VtmnModal/VtmnModal.stories.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      showModal,
+      handleShowModalClick: () => {
+        showModal.value = true;
+      },
+      args,
+    };
+  },
+  template: `
+  <VtmnButton @click="handleShowModalClick()">Display modal</VtmnButton>
+  <VtmnModal v-if="showModal" v-bind="args">
```
Comment (lauthieb) [inline packages/showcases/vue/stories/components/overlays/VtmnModal/VtmnModal.stories.js:30]:
```suggestion
  <VtmnModal v-bind="args" :open="showModal">
```

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks, few changes :)

**Review by lauthieb — APPROVED** [review]

**Review by ThomasLaigneauDecathlon — APPROVED** [review]

**Pull request merged**